src/Clash/CoSim/Simulator.hs

```diff
@@ -353,17 +353,18 @@ parseInput
     -> [SignalStream]
     -> t
     -> [SignalStream]
-parseInput doDup streams x = dup (toSignalStream x) : streams
+parseInput doDup streams t = dup (toSignalStream t) : streams
   where
     dup | doDup     = go
         | otherwise = id
     go (x:xs) = (x:x:go xs)
+    go []     = []
```

Comment:
I think we should just makes this an `error` to retain current behavior. Maybe @christiaanb can comment on why this pattern isn't exhaustive?

Reply:
It wasn't exhaustive because the list comes from a converted `Signal`, so it should have infinite samples. I guess you're right that an error would be better, something like: `Unexpected empty Signal`.

Reply:
Alright. @leonschoorl I think Christiaan's suggestion is sound. Does that work for you?

Reply:
Yes, erroring is probably safer.
I'll change it to that